go-controller/hybrid-overlay/pkg/controller/overlay_windows.go

```diff
@@ -266,30 +266,53 @@ func EnsureExistingNetworkIsValid(networkName string, expectedAddressPrefix stri
 	return nil
 }
 
-func DuplicatePersistentIPRoutes() error {
+// duplicateIPv4Routes duplicates all the IPv4 network routes associated with the physical interface to the host vNIC,
+// where parameters policyStore and destinationPrefix are optional and could be used as filtering criteria.
+// Otherwise, all IPv4 routes will be forwarded from the physical interface to the host vNIC.
+func duplicateIPv4Routes(policyStore, destinationPrefix string) error {
 	shell, err := ps.New(&psBackend.Local{})
 	if err != nil {
 		return err
 	}
 	defer shell.Exit()
 
+	// build filter options for Get-NetRoute command. See https://docs.microsoft.com/en-us/powershell/module/nettcpip/get-netroute
+	var opts string
+	if policyStore != "" {
+		opts += fmt.Sprintf(" -PolicyStore %s", policyStore)
+	}
+	if destinationPrefix != "" {
+		opts += fmt.Sprintf(" -DestinationPrefix \"%s\"", destinationPrefix)
+	}
+
 	script := `
 	# Find physical adapters whose interfaces are bound to a vswitch (i.e. the MAC addresses match)
 	$boundAdapters = (Get-NetAdapter -Physical | where { (Get-NetAdapter -Name "*vEthernet*").MacAddress -eq $_.MacAddress })
 
-	# Forward all the persistent routes associated with the physical interface to the associated vNIC
+	# Forward all the matching routes associated with the physical interface to the associated vNIC
 	foreach ($boundAdapter in $boundAdapters) {
 		$associatedVNic = Get-NetAdapter -Name "*vEthernet*" | where { $_.MacAddress -eq $boundAdapter.MacAddress }
-		$routes = Get-NetRoute -PolicyStore PersistentStore -InterfaceIndex $boundAdapter.IfIndex -ErrorAction SilentlyContinue
+		$routes = Get-NetRoute` + opts + ` -AddressFamily IPv4 -InterfaceIndex $boundAdapter.IfIndex -ErrorAction SilentlyContinue
 		foreach ($route in $routes) {
-			netsh.exe int ipv4 add route interface=$($associatedVNic.ifIndex) prefix=$($route.DestinationPrefix) nexthop=$($route.NextHop) metric=$($route.RouteMetric) store=persistent
+			netsh.exe int ipv4 add route interface=$($associatedVNic.ifIndex) prefix=$($route.DestinationPrefix) nexthop=$($route.NextHop) metric=$($route.RouteMetric) store=$($route.PolicyStore)
 		}
 	}
 	`
 
 	if _, stderr, err := shell.Execute(script + "\r\n\r\n"); err != nil {
-		return fmt.Errorf("falied to refresh the network persistent routes, %v: %v", stderr, err)
+		return fmt.Errorf("falied to duplicate network routes to the associated vNIC, %v: %v", stderr, err)
 	}
 
 	return nil
 }
+
+func DuplicatePersistentIPRoutes() error {
+	return duplicateIPv4Routes("PersistentStore", "")
+}
+
+// DuplicateLinkLocalIPRoutes copies the routes for link-local addresses, that is 169.254.0.0/16 (169.254.0.0
+// through 169.254.255.255) for IPv4, that used to be on the physical network interface to the newly created host vNIC
+// See https://datatracker.ietf.org/doc/html/rfc3927
+func DuplicateLinkLocalIPRoutes() error {
+	return duplicateIPv4Routes("", "169.254.")
```

Comment:
@jrvaldes Does passing `169.254.` as the `-DestinationPrefix` work? See below:
```powershell
PS C:\Users\Administrator> Get-NetRoute  -DestinationPrefix "169.254."                                                  
Get-NetRoute : No MSFT_NetRoute objects found with property 'DestinationPrefix' equal to '169.254.'.  Verify the value of the property and retry.
At line:1 char:1
+ Get-NetRoute  -DestinationPrefix "169.254."
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (169.254.:String) [Get-NetRoute], CimJobException
    + FullyQualifiedErrorId : CmdletizationQuery_NotFound_DestinationPrefix,Get-NetRoute
```
The issue is that `-ErrorAction SilentlyContinue` as it implies will silently continue. End result is that `169.254.0.0/16` are not getting duplicated!

From what I can tell:
```Powershell
PS C:\Users\Administrator> Get-NetRoute  -DestinationPrefix "169.254.169.254/32"

ifIndex DestinationPrefix                              NextHop                                  RouteMetric ifMetric PolicyStore
------- -----------------                              -------                                  ----------- -------- -----------
7       169.254.169.254/32                             10.0.128.1                                        15 15       ActiveStore
```

You need to specify the exact destination prefix for this to work.

Wouldn't using `-PolicyStore ActiveStore`  in addition do the trick?
```golang
func DuplicateActiveStoreIPRoutes() error {
	return duplicateIPv4Routes("ActiveStore", "")
}